src/components/Auth/Consents/CommunicationConsent/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import Trans from 'next-translate/Trans';
+
+import ConsentBody from '../ConsentBody';
+
+import styles from './CommunicationConsent.module.scss';
+
+import ConsentType from '@/types/auth/ConsentType';
+import { logButtonClick } from '@/utils/eventLogger';
+
+type Props = {
+  consentType: ConsentType;
+  isLoading: boolean;
+  onCompleted: (consentType: ConsentType, consented: boolean) => void;
+};
+
+const CommunicationConsent: React.FC<Props> = ({ onCompleted, consentType, isLoading }) => {
+  const onButtonClicked = (consented: boolean) => {
+    logButtonClick(`${consentType}_${consentType}`);
+    onCompleted(consentType, consented);
+  };
+
```

Comment:
I see we're trying to make this dynamic and work for all consent types (which is amazing), however some stuff still are hard-coded. 

1. We need to rename the component to something like `ConsentModal`
2. We need to dynamically get the `i18nKey` using the `consentType` instead of hard-coding `communication`